src/ParallelAlgorithms/Amr/Criteria/DriveToTargetAndOscillate.hpp

```diff
@@ -0,0 +1,122 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <pup.h>
+
+#include "Domain/Amr/Flag.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "ParallelAlgorithms/Amr/Criteria/Criterion.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace amr::Criteria {
+/*!
+ * \brief Refine the grid towards the target resolution and refinement levels in
+ * each dimension and then oscillate about the target.
+ *
+ * \details If the grid is at neither target in a given dimension, the
+ * flag chosen will be in the priority order Split,
+ * IncreaseResolution, DecreaseResolution, Join.
+ *
+ * \note This criterion is primarily for testing the mechanics of refinement.
+ */
+template <size_t Dim>
+class DriveToTargetAndOscillate : public Criterion {
+ public:
+  /// The target resolution
+  struct TargetResolution {
+    using type = std::array<size_t, Dim>;
+    static constexpr Options::String help = {
+        "The target resolution in each dimension."};
+  };
+
+  /// The target resolution
+  struct TargetRefinementLevels {
+    using type = std::array<size_t, Dim>;
+    static constexpr Options::String help = {
+        "The target refinement levels in each dimension."};
+  };
+
+  /// The AMR flags chosen when the target resolution and refinement levels are
+  /// reached
+  struct DecisionAtTarget {
+    using type = std::array<domain::Flag, Dim>;
```

Comment:
Maybe make this an optional so we can drive to the target resolution and stay there? Then you can move the "AndOscillate" from the class name to this option, maybe name "OscillateAtTarget".

Reply:
you can stay at the target by choosing the flags to be `DoNothing`

Reply:
Oh good. Then just move the "oscillate" name here?

Reply:
okay, done and squashed